app/frontend/wallet/shelley-wallet.ts

```diff
@@ -36,6 +36,15 @@ import {
   ShelleyTtl,
   ShelleyWitdrawal,
 } from './shelley/shelley-transaction'
+import {
+  DelegetionHistoryObject,
+  StakePool,
+  StakeDelegation,
+  DelegetionHistoryItemType,
+  StakingReward,
+  RewardWithdrawal,
+} from '../components/pages/delegations/delegationHistoryPage'
```

Comment:
To be honest, I don't really like exporting anything from components to this shelley-wallet. What we could do, is to define these interfaces in `state.ts` as they refer to objects in the state anyway. And export the interfaces from there.

Reply:
I've been thinking about this as well, but i don't like putting it into the `state.ts`, if everything goes there it would end up as really big file eventually. I think that interfaces should be part of relevant file where they are mainly used (or at least as separate file (something like `delegateHistoryInterfaces`?) under same folder hierarchy - `app/frontend/components/delegations/`), similarly to OOP where classes are defined with their typed fields/members/functions they are usually contained in one class (file).

Purpose of these interfaces is to dictate the structure of objects that we want to render in component, so I think it makes sense to keep them within file that uses them. If they happen to be in more components, it would make sense to put it into other common file.